src/lib.rs

```diff
@@ -237,3 +237,60 @@ impl From<Protocol> for c_int {
         p.0
     }
 }
+
+/// Flags for incoming messages.
+///
+/// Flags provide additional information about incoming messages.
+#[cfg(not(target_os = "redox"))]
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub struct RecvFlags(c_int);
+
+#[cfg(not(target_os = "redox"))]
+impl RecvFlags {
+    /// Check if the message terminates a record.
+    ///
+    /// Not all socket types support the notion of records.
+    /// For socket types that do support it (such as [`SEQPACKET`][Type::SEQPACKET]),
+    /// a record is terminated by sending a message with the end-of-record flag set.
+    ///
+    /// On Unix this corresponds to the MSG_EOR flag.
+    #[cfg(unix)]
```

Comment:
Can you move this and `is_out_of_band` to the sys/unix.rs file, similar to e.g. `Domain` extension: https://github.com/alexcrichton/socket2-rs/blob/d653b7b5e0b2438677b721508f4ba5322a9bd81a/src/sys/unix.rs#L98-L110

Reply:
Moved, along with the `Debug` impl.